test/objectid.test.js

```diff
@@ -32,41 +34,75 @@ describe('ObjectID', function() {
   });
 
   it('should convert 24 byte hex string as ObjectID', function() {
-    var ObjectID = db.connector.getDefaultIdType();
-    var str = '52fcef5c0325ace8dcb7a0bd';
-    ObjectID(str).should.be.an.instanceOf(db.ObjectID);
+    var ObjectID = ds.connector.getDefaultIdType();
+    var str = objectIDLikeString;
+    ObjectID(str).should.be.an.instanceOf(ds.ObjectID);
   });
 
   it('should not convert 12 byte string as ObjectID', function() {
-    var ObjectID = db.connector.getDefaultIdType();
+    var ObjectID = ds.connector.getDefaultIdType();
     var str = 'line-by-line';
     ObjectID(str).should.be.equal(str);
   });
 
   it('should keep mongodb ObjectID as is', function() {
-    var ObjectID = db.connector.getDefaultIdType();
-    var id = new db.ObjectID();
-    ObjectID(id).should.be.an.instanceOf(db.ObjectID);
+    var ObjectID = ds.connector.getDefaultIdType();
+    var id = new ds.ObjectID();
+    ObjectID(id).should.be.an.instanceOf(ds.ObjectID);
   });
 
   it('should keep non-string id as it', function() {
-    var ObjectID = db.connector.getDefaultIdType();
+    var ObjectID = ds.connector.getDefaultIdType();
     var id = 123;
     ObjectID(id).should.be.equal(123);
   });
 
-  it('coerces ObjectID', function() {
-    const coercedId = db.connector.isObjectIDProperty('Book', {}, '52fcef5c0325ace8dcb7a0bd');
-    coercedId.should.be.True();
-  });
+  context('strictObjectIDCoercion', function () {
+    context('set to false (default)', function () {
+
+      beforeEach(function(done) {
+        Book.deleteAll(done);
+      });
+
+      it('it should coerce to ObjectID', async function() {
+        Book = ds.createModel(
+          'book',
+          {
+            xid: {type: String}
+          }
+        );
+        const book = await Book.create({xid: objectIDLikeString});
+        book.xid.should.be.an.instanceOf(ds.ObjectID);
```

Comment:
@b-admike any idea why this fails despite the line 2112 being executed in `/lib/mongodb.js`?

Reply:
Yea Janny's right. I had to query the db myself and get the raw results back (see https://github.com/strongloop/loopback-connector-mongodb/blob/master/test/decimal.test.js#L468-L470) to make the assertions.

Reply:
I see. So we have a sort of confusing behavior, here. In fact, I remember now, I was thinking of coercing in the properties the result too.

Now, the question is. Should the query result preserve the ObjectID type? In my opinion, we should.

@jannyHou @b-admike @bajtos @raymondfeng

Reply:
I don't have a strong opinion. I agree the current behavior is confusing. To me, it's important to leave any fixes out of scope of this pull request. Considering how much time we have already spend on fixing the MongoDB ObjectID coercion issue, I am reluctant to get distracted by other improvements. Let's open a new GitHub issue to describe the problem so that we can properly plan the time needed to fix it; and rework the tests to accommodate limitations of the current implementation.

Reply:
Agreed on deferring this to another issue. In terms of the behaviour, the query result is actually coerced for decimal types. It's just juggler which converts it back to string values. I think this is because, say, for explorer, we need to have it displayed as a string value.